modules/markup/markdown/math/inline_parser.go

```diff
@@ -45,6 +45,10 @@ func isPunctuation(b byte) bool {
 	return b == '.' || b == '!' || b == '?' || b == ',' || b == ';' || b == ':'
 }
 
+func isBracket(b byte) bool {
+	return b == '(' || b == ')' || b == '[' || b == ']' || b == '{' || b == '}'
```

Comment:
Overall LGTM.

GitHub only supports `($ $)`, so I would suggest to follow its behavior until there is a clear requirement for other cases.

![image](https://github.com/go-gitea/gitea/assets/2114189/04ffbabc-3bba-4f8d-b97e-70229e52816e)